src/main/java/edu/hw1/Task5.java

```diff
@@ -0,0 +1,66 @@
+package edu.hw1;
+
+import java.util.ArrayList;
+
+public class Task5 {
+
+    boolean isPalindromeDescendant(long parent) {
+        String parentStr = String.valueOf(parent);
+
+        return switch (parentStr.length()) {
+            case 1 -> false;
+            case 2 -> parentStr.charAt(0) == parentStr.charAt(1);
+            default -> {
+                String parentStrFromEnd = headToLegs(parentStr);
+                yield parentStrFromEnd.equals(parentStr)
+                    || (checkDescForPalindrome(parent) && parentStrFromEnd.charAt(0) != '0');
+            }
+        };
+    }
+
+    private String headToLegs(String toDo) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = toDo.length() - 1; i >= 0; i--) {
+            sb.append(toDo.charAt(i));
+        }
+        return sb.toString();
+    }
+
+    private ArrayList<Integer> makeIntsFromNumber(long number) {
```

Comment:
`BUG` даже в приватных функциях не возвращай  ArrayList -- надо всегда возвращать List
```suggestion
    private List<Integer> makeIntsFromNumber(long number) {
```

Reply:
Это для гибкости или?

Reply:
Если в двух словах, то  для гибкости https://www.baeldung.com/java-list-vs-arraylist . А по сути это отголоски D-dependency inversion из Solid'a. Понимаю, что пока вообще непонятно зачем, но просто возьми за привычку где возможно вместо ArrayList<> писать list

Reply:
Понятно. Без SOLID да, никак, ни на одну работу не возьмут)